### ISSUE TITLE
the current state of the project and info

### ISSUE THREAD

**Issue by AleXoundOS** [ShieldBattery/ShieldBattery#399]
Since the slowdown of the development and indefinite future of the project some questions arise.

It would be nice to update the README with the following information in detail:
* The current state of the project.
* How does it compare with official updated servers? Therefore, which features are unique? Comparison matrix.
* How well does it integrate with SC:R client?
* Architecture of the system. Descriptions of folders is not the same.

Also, in case the project ceases development before becoming mature, it can be useful to document how many of technical aspects were implemented.

**Comment by 2Pacalypse-** [comment]
I don't feel like updating the README at this point, but I'll write some things here.
#### 1. The current state of the project
TLDR: The current state of the project is despair and disbelief :). More seriously, the development is at an indefinite halt. I, personally, still do a few things here and there, mainly upgrading the dependencies to keep it up to date, but it's unlikely it will become much more than that in the foreseeable future.

Longer version: https://docs.google.com/document/d/1rASPAOTD1L6ClyHQCZszdeMMxksfuHNmB2sBKrWSSws/edit?usp=sharing

This was an internal document written a couple of months before SC:R release, so it might contain a few inaccuracies now. It was not meant to be read by the public, but enough time has passed that I see no downsides in posting it.
#### 2. SC:R comparison
I honestly don't feel like writing a feature-to-feature comparison with SC:R, since there are a bunch of features in ShieldBattery that are either hidden behind feature-flags (eg. support for multiple chat channels) and some that are like 95% finished, but not released yet (eg. auto-matchmaking support). There are also features that both the SC:R and ShieldBattery have (eg. the network solution for hosting problems), but I'm pretty sure the ShieldBattery does it better.

However, there are two things (not features, more like philosophies) that I'd like to point out as the main difference between ShieldBattery and SC:R. The first one is ShieldBattery's focus on the social experience of its users. It's no accident that the sleek interface and chat channels were one of the main features in ShieldBattery, as the idea was always to have something like Discord is today (note: this was before Discord was even a thing), but specifically tailored for StarCraft community and seamlessly integrated with the StarCraft, the game. The social experience in SC:R leaves a lot to be desired; not only does it not support multiple chat channels, but there are also segregated regional gateways, which makes socializing with people a real hurdle.

The second thing might sound a bit cheesy and irrelevant, but it's equally important imo. ShieldBattery was developed and directed by long-time members of the Brood War community, who have a long history of building its own solutions to problems. While taking the community's input into account is important, SC:R's development team are overly relying on community hand-holding them through every decision, sometimes with [embarrassing results](https://www.teamliquid.net/forum/brood-war/542230-grant-d-making-decisions-based-on-twitter-poll).
#### 3. SC:R integration
ShieldBattery currently only works with StarCraft 1.16.1 version. However, there is a downgrade system in place, which makes it possible to downgrade newer versions of StarCraft down to 1.16.1. So people don't need to track down and download version 1.16.1 on their own. If they only have SC:R installed, they should still be able to play on ShieldBattery, which will automatically downgrade their version to 1.16.1 on the fly (meaning, their original SC:R version will remain  untouched). Note: this system requires admin to create diff-patches of StarCraft binaries and upload them to the server, which hasn't been done for a while now; but the system should still work, I think.

Updating the ShieldBattery to fully support SC:R requires quite a bit of work, which is actually one of the main reasons the project is not being developed at the moment. While it's not impossible, I'm not sure when, if ever, this will be done.
#### 4. Architecture
This is actually a really good idea. Since this post is already way too long, and describing the ShieldBattery's architecture could take a few pages all on its own, I'll try to write a separate post, or a google document, which describes it in great detail. It might take me a while to finish that though, but I hope I will finish it sooner or later.

If you can't wait, then here's a *really* condensed version of it :)
```
Electron Application <- ws -> Node.js Addons -> Node.js(shieldbattery.dll) -> StarCraft.exe
          ^
          |
       https/wss
          |
          v
Node.js server -> PostgreSQL Database
          ^
          |
       https/wss
          |
          v
Website Application
```

**Comment by AleXoundOS** [comment]
Thank you for the response and the progress the project has done since then!
